arrivals/src/main/scala/com/pagerduty/arrivals/impl/ArrivalsServer.scala

```diff
@@ -0,0 +1,110 @@
+import com.pagerduty.arrivals.impl.aggregator.{
+  AggregatorController,
+  AggregatorRequestHandler
+}
+import com.pagerduty.arrivals.impl.auth.{
+  RequestAuthenticator,
+  RequireAuthentication
+}
+import com.pagerduty.arrivals.impl.authproxy.{
+  AuthProxyController,
+  AuthProxyRequestHandler
+}
+import com.pagerduty.arrivals.impl.headerauth.HeaderAuthenticator
+import com.pagerduty.arrivals.impl.proxy.{
+  HttpProxy,
+  ProxyController,
+  ProxyRequestHandler
+}
+import com.pagerduty.metrics.{Metrics, NullMetrics}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+abstract class ArrivalsServer[AddressingConfig, AuthConfig <: HeaderAuthConfig](
+    addressingConfig: AddressingConfig,
+    headerAuthConfig: AuthConfig,
+    listenInterface: String = "0.0.0.0",
+    listenPort: Int = 8080,
+    entityConsumptionTimeout: FiniteDuration = 20.seconds)(
+    implicit actorSystem: ActorSystem,
+    val materializer: Materializer,
+    metrics: Metrics = NullMetrics)
+    extends ProxyController[AddressingConfig]
+    with AuthProxyController[AuthConfig, AddressingConfig]
+    with AggregatorController[AuthConfig, AddressingConfig] { outer =>
+
+  def routes: Route
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  logger.info("Starting HttpServer...")
+  val authConfig = headerAuthConfig
+  implicit val executionContext = actorSystem.dispatcher
+
+  val httpProxy =
+    new HttpProxy(addressingConfig, Http().singleRequest(_))
+
+  val proxyRequestHandler =
+    new ProxyRequestHandler[AddressingConfig] {
+      val executionContext = outer.executionContext
+    }
+
+  val authProxyRequestHandler =
+    new AuthProxyRequestHandler[AddressingConfig, AuthConfig#AuthData] {
+      val executionContext = outer.executionContext
+    }
+
+  val aggregatorRequestHandler =
+    new AggregatorRequestHandler[AddressingConfig, AuthConfig#AuthData] {
+      val executionContext = outer.executionContext
+    }
+
+  val requestAuthenticator = new RequestAuthenticator {
+    val executionContext = outer.executionContext
+    val metrics = outer.metrics
+  }
+
+  val requireAuthentication = new RequireAuthentication {
+    val requestAuthenticator = outer.requestAuthenticator
+  }
+
+  val headerAuthenticator = new HeaderAuthenticator {
+    val requestAuthenticator = outer.requestAuthenticator
+  }
+
+  logger.info(
+    s"Akka-HTTP binding to port: $listenPort and interface: $listenInterface...")
+  private val bindingFuture =
+    Http(actorSystem).bindAndHandle(routes, listenInterface, listenPort)(
+      materializer)
+
+  //   for simplicity, block until the HTTP server is actually started
+  private val tryBinding = Try(Await.result(bindingFuture, 10.seconds))
+
+  private val binding = tryBinding match {
+    case Success(b) =>
+      logger.info(
+        s"Successfully bound to port: ${b.localAddress.getPort} on interface: $listenInterface")
```

Comment:
```suggestion
        s"Successfully bound to port: ${b.localAddress.getPort} on interface: ${b.localAddress.getAddress}")
```
seems more consistent? or just
```suggestion
        s"Successfully bound to ${b.localAddress}")
```